bluemira/codes/_freecadapi.py

```diff
@@ -1402,9 +1402,21 @@ def save_cad(
         ) from imp_err
 
     if not os.path.exists(filename):
+        mesg = f"{filename} not created, filetype not written by FreeCAD."
+        if formatt is CADFileType.IFC_BIM:
+            mesg += " FreeCAD requires `ifcopenshell` to save in this format."
+        elif formatt is CADFileType.DAE:
+            mesg += " FreeCAD requires `pycollada` to save in this format."
+        elif formatt is CADFileType.IFC_BIM_JSON:
+            mesg += (
+                " FreeCAD requires `ifcopenshell` and"
+                " IFCJSON module to save in this format."
+            )
+        elif formatt is CADFileType.AUTOCAD:
+            mesg += " FreeCAD requires `LibreDWG` to save in this format."
+
         raise FileNotFoundError(
-            f"{filename} not created, filetype not written by FreeCAD."
-            f"Possibly no object compatible with '{formatt.value}'"
+            f"{mesg} Not able to save object with format: '{formatt.value}'"
```

Comment:
Is `FileNotFoundError` the right error to raise here? When writing, I would only really expect a `FileNotFoundError` if we were trying to write a file to a directory that doesn't exist.

Not really sure what the best error to raise would be though, maybe a `CADError`, if we have that or something similar?

Reply:
We have `FreeCADError` just above, I've put that in.